showcase/src/app/shared/effects/stark-error-handler.effects.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable, Injector } from "@angular/core";
+import { Actions, Effect, ofType } from "@ngrx/effects";
+import { Observable } from "rxjs";
+import { map, take } from "rxjs/operators";
+
+import { STARK_TOAST_NOTIFICATION_SERVICE, StarkMessageType, StarkToastNotificationService } from "@nationalbankbelgium/stark-ui";
+import { StarkErrorHandlerActionTypes, StarkUnhandledError } from "@nationalbankbelgium/stark-core";
+
+/**
+ * Unique Id of the displayed toaster
+ */
+const _uniqueId: Function = require("lodash/uniqueId");
+
+/**
+ * This class is used to determine what to do with an error
+ */
+@Injectable()
+export class StarkErrorHandlerEffects {
+	private _starkToastNotificationService: StarkToastNotificationService;
+
+	/**
+	 * Class constructor
+	 * @param actions$ - the action to perform
+	 * @param injector - the injector of the class
+	 */
+	public constructor(private actions$: Actions, private injector: Injector) {}
+
+	@Effect({ dispatch: false })
+	public setToastMessage$(): Observable<void> {
+		return this.actions$.pipe(
+			ofType<StarkUnhandledError>(StarkErrorHandlerActionTypes.UNHANDLED_ERROR),
+			map((action: StarkUnhandledError) => {
+				this.toastNotificationService
+					.show({
+						id: _uniqueId(),
+						type: StarkMessageType.ERROR,
+						key: action.error.toString(),
+						code: "Unhandled error - no code"
+					})
+					.pipe(take(1))
```

Comment:
After testing the proposed solution, it seems that the `take` operator is not needed anymore.